### DISCUSSION TITLE
Apply different intensity transforms per channel

### DISCUSSION THREAD

**Discussion by jpcenteno80** [Project-MONAI/MONAI#2464] [General]
Hi, I am working with CT and would like to apply a different `a_min`, `a_max` using `ScaleIntensityRange` per channel. For example,  load input image with shape `1,64,64,64` (channel,spatial_dim,spatial_dim,spatial_dim), apply `RepeatChannel` transform to end up with shape `2,64,64,64` and then apply `ScaleIntensityRange` with `(a_min=-300,a_max=300)` for one channel and `(a_min=130,a_max=3000)` for the other channel. I would like this to be part of a dictionary style transform chain. Is this possible with your current transforms? Thank you!

**Comment by jpcenteno80** [comment]
Found the solution using these transforms after loading the image/label and before creating a tensor inside a `Compose` object:
```
           ...
            RepeatChanneld('image', repeats=2),
            SplitChanneld('image', channel_dim=0),
            ScaleIntensityRanged('image_0', a_min=heart_range[0], a_max=heart_range[1],
                                b_min=0.0, b_max=1.0, clip=True),
            ScaleIntensityRanged('image_1', a_min=bone_range[0], a_max=bone_range[1],
                                b_min=0.0, b_max=1.0, clip=True),
            ConcatItemsd(['image_0', 'image_1'], name='image', dim=0),
            DeleteItemsd(['image_0', 'image_1']),
            ...
```

**Reply by Nic-Ma** [reply]
Hi @jpcenteno80 ,

Thanks for your interest here.
Actually, I would suggest you to use the `CopyItemsd` transform instead of `RepeatChanneld` transform.
After copying, you can apply `ScaleIntensityranged` on different keys and concat the results.

Thanks.
